### PR TITLE
No org for SSO users

### PR DESCRIPTION
**Pull request by svenefftinge**
## Description

New users automatically get the migrated flag (property to be removed once migration is done) and only get an org if they are not owned by an org (SSO)

## Related Issue(s)

Fixes WEB-272

## How to test

login to the oreview env and verify that you don't have an org.

#### Preview status

gitpod:summary

## Build Options:

- [ ] /werft with-werft
      Run the build with werft instead of GHA
- [ ] leeway-no-cache
- [ ] /werft no-test
      Run Leeway with `--dont-test`

<details>
<summary>Publish Options</summary>

- [ ] /werft publish-to-npm
- [ ] /werft publish-to-jb-marketplace
</details>

<details>
<summary>Installer Options</summary>

- [ ] analytics=segment
- [x] with-dedicated-emulation
- [ ] with-ws-manager-mk2
- [ ] workspace-feature-flags
  Add desired feature flags to the end of the line above, space separated
</details>

#### Preview Environment Options:
- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [x] /werft with-preview
- [ ] /werft with-large-vm
- [ ] /werft with-gce-vm
      If enabled this will create the environment on GCE infra
- [ ] with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`

/hold

### CODE REVIEW

File: components/server/src/user/user-service.ts [hunk 1]
```diff
@@ -128,21 +128,18 @@ export class UserService {
         // state. This measure of prevention is considered in the period deleter as well.
         newUser.identities.push({ ...identity, deleted: false });
         this.handleNewUser(newUser, isFirstUser);
+        if (!!newUser.organizationId) {
```
Comment (geropl) [inline components/server/src/user/user-service.ts:131]: `!newUser.organizationId` ?

File: components/gitpod-db/src/user-to-team-migration-service.ts [hunk 2]
```diff
@@ -144,7 +144,6 @@ export class UserToTeamMigrationService {
     }
 
     async needsMigration(user: User): Promise<boolean> {
-        const teams = await this.teamDB.findTeamsByUser(user.id);
-        return teams.length === 0 || !user.additionalData?.isMigratedToTeamOnlyAttribution;
+        return !user.additionalData?.isMigratedToTeamOnlyAttribution;
```
Comment (geropl) [inline components/gitpod-db/src/user-to-team-migration-service.ts:147]: :+1:

Comment (svenefftinge) [reply components/gitpod-db/src/user-to-team-migration-service.ts:147]: FWIW: This case can happen in situations where a user deletes their last org or a user is kicked out of their last org.
In those cases we need to require creating an org (or use an invite link to join one) so I thought we could use the "user has no orgs" condition in general to show the first step of the "dedicated onboarding flow".

Comment (geropl) [reply components/gitpod-db/src/user-to-team-migration-service.ts:147]: > I thought we could use the "user has no orgs" condition in general to show the first step of the "dedicated onboarding flow"

Why not show the `/org/new` view? :thinking:

Comment (svenefftinge) [reply components/gitpod-db/src/user-to-team-migration-service.ts:147]: We can of course. I just wished we wouldn't have two different ways and UIs to create an org, but I might have expressed that preference already 😇

**Comment by svenefftinge** [comment]
closing this as I opted for a less invasive solution

**Pull request closed without merging**